### DISCUSSION TITLE
Start trilium with scripts/widgets disabled?

### DISCUSSION THREAD

**Discussion by nicollasricas** [TriliumNext/Trilium#2892] [Q&A] [answered]
Hey @zadam,

Is there a way to start trillium with scripts or widgets disabled? I f*** up some scripts, and now the UI doesn't even render. It was a widget that I was implementing.

I accidentally pressed CTRL + R when writing the script.

![image](https://user-images.githubusercontent.com/7860985/171780854-d5bde6eb-7bee-4a97-902c-16f4821cafcb.png)

**Comment by zadam** [comment, marked as answer]
Hi, yes, there's "safe mode" for this: https://github.com/zadam/trilium/wiki/Troubleshooting#broken-script-prevents-application-startup

In such cases you can start Trilium in "safe mode" which will not execute any custom scripts:

```
TRILIUM_SAFE_MODE=true ./trilium
```

Depending on the Trilium distribution, there are also ready made scripts for this: `trilium-safe-mode.bat` and `trilium-safe-mode.sh`.
